.github/workflows/upload-test-stats.yml

```diff
@@ -87,6 +87,7 @@ jobs:
     if: ${{ always() }}
     runs-on: [self-hosted, linux.2xlarge]
     continue-on-error: true
+    environment: update-commit-hash
```

Comment:
Please note update-commit-hash restricted to main branch only. if this job needs to be run on different branch let me know. We would need to create new env

Reply:
upload test stats is only run on workflow_run trigger, which is always from the main branch, should it should be fine

As an aside, we might want to consider using a different environment name because the current one seems very specific haha